src/System.Net.Http/src/System/Net/Http/StreamContent.cs

```diff
@@ -12,19 +11,30 @@ namespace System.Net.Http
 {
     public class StreamContent : HttpContent
     {
-        private const int defaultBufferSize = 4096;
+        private const int DefaultBufferSize = 4096;
 
         private Stream _content;
+        private CancellationToken _cancellationToken;
         private int _bufferSize;
         private bool _contentConsumed;
         private long _start;
 
         public StreamContent(Stream content)
-            : this(content, defaultBufferSize)
+            : this(content, DefaultBufferSize)
         {
         }
 
         public StreamContent(Stream content, int bufferSize)
+            : this(content, bufferSize, CancellationToken.None)
+        {
+        }
+
+        internal StreamContent(Stream content, CancellationToken cancellationToken)
```

Comment:
In terms of response body buffering, the *Content object is always an `StreamContent` object.  So, we have total control of that.

Developers create custom HttpContent-derived classes for the purpose of handling request body content. I.e., to make it easier for them to bundle up that content to POST, instead of having to serialize it themselves as ByteArrayContent, for example.

But this scenario of being able to cancel response body buffering only uses the StreamContent class which is part of the System.Net.Http library.